### PR TITLE
Fixed InterRangeType multiplication/division of maximas by a negative constant

### PR DESCRIPTION
**Pull request by staabm**
As discussed in https://github.com/phpstan/phpstan-src/pull/669#discussion_r706811109 maxima need to be inverse when multiplying with a negative constant operand

refs https://github.com/phpstan/phpstan/issues/5614

see https://3v4l.org/AUHH7

### CODE REVIEW

File: tests/PHPStan/Analyser/data/integer-range-types.php [hunk 1]
```diff
@@ -232,8 +232,8 @@ public function math($i, $j, $z, $pi, $r1, $r2, $rMin, $rMax, $x, $y) {
 		assertType('int<-2, 9>|int<21, 30>', $r1 - $z);
 		assertType('int<-200, -20>|int<1, 30>', $r1 * $z);
 		assertType('float|int<0, 10>', $r1 / $z);
-		assertType('int<min, 15>', $rMin * $z);
-		assertType('int<-100, max>', $rMax * $z);
+		assertType('int', $rMin * $z);
```
Comment (staabm) [inline tests/PHPStan/Analyser/data/integer-range-types.php:235]: Multiplication by $z leads to a integer range with no upper bound unioned with a range without a lower bound.

**Comment by staabm** [comment]
Remaining failures seem unrelated - otherwise good to go

File: src/Analyser/MutatingScope.php [hunk 2]
```diff
@@ -1021,12 +1021,7 @@ private function resolveType(Expr $node): Type
 			}
 
 			if ($type instanceof IntegerRangeType) {
-				$negativeRange = $this->resolveType(new Node\Expr\BinaryOp\Mul($node->expr, new LNumber(-1)));
-
-				if ( $negativeRange instanceof IntegerRangeType && ($negativeRange->getMin() === null || $negativeRange->getMax() === null)) {
```
Comment (staabm) [inline src/Analyser/MutatingScope.php:1026]: This check was moved into the integerRangeMath method and therefore fixes multiplication with a negative number for the general case

**Comment by staabm** [comment]
I just added the same patch for division, which initially only was made for multiplication.

we could make this even more complete by supporting [multiplication/division with negative intervals](https://www.wolframalpha.com/input/?i=interval%5B%E2%88%9E%2C5%5D+*+interval%5B-2%2C+-5%5D), e.g. `int<min, 5> * int<-2, -5>` but atm this looks like a rare edge case.

the PR as is already fixes a few glitches, therefore I would like to land it as is.

**Comment by ondrejmirtes** [comment]
Thank you!

**Pull request merged**